### ISSUE TITLE
The placement of focus after opening dropdowns

### ISSUE THREAD

**Issue by Comandeer** [ckeditor/ckeditor5#11838]
## 📝 Provide detailed reproduction steps (if any)

1. Open https://ckeditor.com/docs/ckeditor5/latest/examples/builds/document-editor.html
2. Focus the text alignment button using the keyboard.
3. Press <kbd>Space</kbd>.

### ✔️ Expected result

The focus is moved to the first child (first focusable element) inside the open dropdown.

### ❌ Actual result

The focus stays on the pressed button.

However, some dropdowns (e.g. media embed one) moves focus to the first focusable element upon opening.

### ❓ Possible solution

Make all dropdowns consistent and move focus to the first focusable element. See [ARIA Practices](https://www.w3.org/WAI/ARIA/apg/example-index/menu-button/menu-button-actions-active-descendant.html) for more info.

## 📃 Other details

* Browser: all
* OS: all
* Installed CKEditor plugins: N/A

See also #2000.

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by mlewand** [comment]
Question: What should get focused after opening the dropdown?

Should it be the first item or should it be the first active option?

*   CKE4 puts the focus on the active/"pressed" option of the dropdown (an example is [font size dropdown](https://ckeditor.com/docs/ckeditor4/latest/examples/fullpreset.html)).
*   Gdocs is inconsistent:
    *   Font family is placing the focus in the active option.
    *   Font color dropdown and list style places the focus _god knows where_ - only after the first key press it comes into the dropdown content.
*   Other popular WYSIWYG editors place the focus on the first item always.

So unless the best practices doesn't explicitly say what should be focused we can go either way (whatever will be quicker to implement).

**Comment by mlewand** [comment]
> According to [ARIA 1.2 specification](https://www.w3.org/TR/wai-aria-1.2/#managingfocus) and [ARIA Practices](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#x6-6-keyboard-navigation-inside-components), it depends on the type of the dropdown. For [menus](https://www.w3.org/WAI/ARIA/apg/example-index/menu-button/menu-button-actions.html) (so basically lists of actions), it should be the first element inside them. AFAIK we don't have typical menus in CKE5. For [comboboxes](https://www.w3.org/WAI/ARIA/apg/example-index/combobox/combobox-select-only.html), it should generally be the selected element (or, if nothing is selected, the first one). Most of our dropdowns are comboboxes so IMO it's the way to go.

Thanks for summing it up @Comandeer - this sounds reasonable.

**Comment by mlewand** [comment]
*   We need to ensure that all the dropdowns work after it (solid QA testing). For example table dropdown will not work.
*   Our solution should not destroy (crash) existing implementation.
    *   It would be nice to log some warning if a dropdown doesn't have anything focusable.
*   As for focusing active/selected item we could implement this logic in https://ckeditor.com/docs/ckeditor5/latest/api/module_ui_dropdown_utils.html#function-addListToDropdown
